// @corva/eslint-config-browser/eslint/index.js
module.exports = {
  extends: ['airbnb', 'prettier'],
  plugins: ['prettier', 'react-hooks'],
  parser: '@babel/eslint-parser',
  rules: {
    'class-methods-use-this': 'off',
    'comma-dangle': ['error', 'only-multiline'],
    'implicit-arrow-linebreak': 'off',
    'import/extensions': 'off',
    'import/no-unresolved': 'off',
    'import/prefer-default-export': 'off',
    // TODO: Setup linebreaks from CRLF to LF on Windows machines
    'linebreak-style': 'off',
    'max-len': 'off',
    'no-console': [
      'warn',
      {
        allow: ['error'],
      },
    ],
    'no-else-return': 'off',
    'no-extend-native': [
      'error',
      {
        exceptions: ['Number'],
      },
    ],
    'no-shadow': 'off',
    'no-underscore-dangle': ['error', { allow: ['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] }],
    'no-unused-vars': [
      'error',
      {
        args: 'after-used',
        ignoreRestSiblings: true,
        argsIgnorePattern: '^_',
      },
    ],
    'operator-linebreak': 'off',
    'react-hooks/exhaustive-deps': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'react/button-has-type': 'off',
    'react/destructuring-assignment': 'off',
    'react/jsx-curly-newline': 'off',
    'react/jsx-filename-extension': ['error', { extensions: ['.js', '.jsx'] }],
    'react/jsx-one-expression-per-line': 'off',
    'react/jsx-props-no-spreading': 'off',
    'react/jsx-tag-spacing': 'off',
    'react/jsx-uses-react': 'off',
    'react/jsx-wrap-multilines': 'off',
    'react/no-access-state-in-setstate': 'off',
    'react/no-did-update-set-state': 'off',
    'react/no-this-in-sfc': 'off',
    'react/prop-types': ['error', { ignore: ['router', 'params', 'location', 'classes'] }],
    'react/react-in-jsx-scope': 'off',
    'react/sort-comp': 'off',
    camelcase: ['error', { allow: ['^UNSAFE_'], properties: 'never', ignoreDestructuring: true }],
    // NOTE: Accessibility rules
    'jsx-a11y/anchor-is-valid': [
      'error',
      {
        // NOTE: Waits for 'to' attribute instead of 'href' in Link component
        components: ['Link'],
        specialLink: ['to'],
      },
    ],
    'jsx-a11y/click-events-have-key-events': 'off',
    'jsx-a11y/control-has-associated-label': 'off',
    'jsx-a11y/no-static-element-interactions': 'off',
  },
  env: {
    browser: true,
  },
  globals: {
    // Specs
    afterEach: true,
    beforeEach: true,
    describe: true,
    expect: true,
    it: true,
    jest: true,
  },
};
